packages/xo-web/src/xo-app/vm/tab-disks.js

```diff
@@ -790,13 +894,19 @@ export default class TabDisks extends Component {
           <Col>
             {newDisk && (
               <div>
-                <NewDisk vm={vm} onClose={this._toggleNewDisk} />
+                <NewDisk
+                  hostSrShouldBeOn={hostSrShouldBeOn}
```

Comment:
We only test the compatibility when the host is defined `hostSrShouldBeOn === '' ? undefined : hostSrShouldBeOn`

Reply:
I avoid condition in render

Reply:
here just to get host's name and display it in an error message when SR is illegal.

Reply:
Its just a simple condition it not cost a lot of resources. But we'll gain on the code comprehension. The parent need this value to display or not a message but the child not needed, it just need to know if it should compare the sr host with the given host or not.